components/event-simulator/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/CSVFileDeployer.java

```diff
@@ -42,92 +40,76 @@ public void init(ConfigurationContext configurationContext) {
     }
 
     public void deploy(DeploymentFileData deploymentFileData) throws DeploymentException {
-
         String path = deploymentFileData.getAbsolutePath();
-
         try {
             processDeploy(deploymentFileData);
 
         } catch (Exception e) {
             throw new DeploymentException("CSV file not deployed and in inactive state :  " + new File(path).getName(), e);
         }
+    }
 
+    @Override
+    public void undeploy(String filePath) throws DeploymentException {
+        processUndeploy(filePath);
```

Comment:
@dilini-muthumala 
Since this is the point the code enters our deployer, shall we catch any exceptions that are coming from our code and wrap it inside a deployment exception. Please see the convention that has been approached in other deployers such as EventReceiverDeployer, EventPublisherDeployer for deploy and undeploy methods.